src/realm/query_expression.hpp

```diff
@@ -1521,7 +1596,11 @@ class Value : public ValueBase, public Subexpr2<T> {
             // Compare values one-by-one (one value is one row; no link lists)
             size_t min = minimum(left->ValueBase::m_values, right->ValueBase::m_values);
             for (size_t m = 0; m < min; m++) {
-
+                if constexpr (std::is_same_v<T, Mixed>) {
+                    if (!Mixed::is_comparable(left->m_storage[0], right->m_storage[m])) {
```

Comment:
I don't think we can do this optimization here (and again below); at least I would expect operator NotEqual to return a match on a Mixed that has a non-comparable type

Reply:
Your are right - this construct is from my very first implementation.